third_party/bigframes_vendored/pandas/core/computation/engines.py

```diff
@@ -0,0 +1,121 @@
+# Contains code from https://github.com/pandas-dev/pandas/blob/main/pandas/core/computation/engines.py
+"""
+Engine classes for :func:`~pandas.eval`
+"""
+from __future__ import annotations
+
+import abc
+from typing import TYPE_CHECKING
+
+from bigframes_vendored.pandas.core.computation.align import (
+    align_terms,
+    reconstruct_object,
+)
+from bigframes_vendored.pandas.core.computation.ops import MATHOPS, REDUCTIONS
+from pandas.errors import NumExprClobberingError
+from pandas.io.formats import printing
+
+if TYPE_CHECKING:
+    from bigframes_vendored.pandas.core.computation.expr import Expr
+
+_ne_builtins = frozenset(MATHOPS + REDUCTIONS)
+
+
+def _check_ne_builtin_clash(expr: Expr) -> None:
+    """
+    Attempt to prevent foot-shooting in a helpful way.
+
+    Parameters
+    ----------
+    expr : Expr
+        Terms can contain
+    """
+    names = expr.names
+    overlap = names & _ne_builtins
+
+    if overlap:
+        s = ", ".join([repr(x) for x in overlap])
+        raise NumExprClobberingError(
```

Comment:
I imagine we don't actually use the NumExpr package (https://github.com/pydata/numexpr) right? Maybe we can delete this function?

Reply:
I wonder how much the "python" engine has changed over the years? If it's been pretty stable, then I think there's probably more benefit in cutting the vendored code to just the "python" engine code. If it's pretty actively changing, then keeping this file as close to as in sync with upstream makes more sense.

Reply:
Seems the python engine itself is pretty stable. Went ahead and removed remaining NumExpr stuff